inference/src/basics.rs

```diff
@@ -116,6 +125,40 @@ impl<'a> Core<'a> {
     }
 }
 
+/// Maintains a set of [`SmtPid`]'s which can be canceled whenever necessary.
+/// Composed of a `bool` which tracks whether the set has been canceled, followed by the
+/// [`SmtPid`]'s of the solvers it contains.
+pub struct SolverPids(RwLock<(bool, Vec<SmtPid>)>);
```

Comment:
Could you switch to a regular `Mutex` here? There's no need for a RwLock.

Reply:
You're right, done. I previously had a function for checking whether cancellation has already occurred, which could be done with just `read`, but that use case ended up being dropped.